src/common/abstractPromote.ts

```diff
@@ -48,20 +61,58 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
 
   protected async executePromotion(): Promise<PromotePipelineResult> {
     validateTestFlags(this.flags['test-level'], this.flags.tests);
-    const doceOrg: Org = await Org.create({ aliasOrUsername: this.flags['devops-center-username']?.getUsername() });
+    const doceOrg: Org = this.flags['devops-center-username'] as Org;
     const pipelineStage: PipelineStage = await fetchAndValidatePipelineStage(
       doceOrg,
       this.flags['devops-center-project-name'],
       this.flags['branch-name']
     );
-    this.computeTargetStageId(pipelineStage);
+    this.projectId = pipelineStage.sf_devops__Pipeline__r.sf_devops__Project__c;
+    this.computeSourceStageId(pipelineStage);
+    const asyncOperationId: string = await this.requestPromotion(doceOrg);
 
-    // hardcoded value so it compiles until main logic is implemented
-    return { status: 'status' };
+    if (this.flags.async) {
+      await DeployPipelineCache.set(asyncOperationId, {});
+      // TODO display async message
+    }
+
+    // TODO: move this to logger service
+    this.log(`Job ID: ${asyncOperationId}`);
+
+    return { jobId: asyncOperationId };
+  }
+
+  private async requestPromotion(targetOrg: Org): Promise<string> {
+    const req: HttpRequest = {
+      method: 'POST',
+      url: `${REST_PROMOTE_BASE_URL as string}${this.projectId}/pipelineName/${this.sourceStageId}`,
+      body: JSON.stringify({
+        changeBundleName: this.flags['bundle-version-name'],
+        promoteOptions: this.buildPromoteOptions(),
+      }),
+    };
+    return targetOrg.getConnection().request(req);
+  }
+
+  private buildPromoteOptions(): Partial<PromoteOptions> {
+    // set base promote options
+    this.deployOptions = {
+      fullDeploy: this.flags['deploy-all'],
+      testLevel: this.flags['test-level'] ?? 'Default',
+      runTests: this.flags['tests'] ? this.flags['tests'].join(',') : undefined,
+    };
+    // ask concrete implementation to add more promote options if needed
+    this.addPromoteOptions();
+    return this.deployOptions;
```

Comment:
Why is this stored on the instance and returned?  Do we need it after the promotion is requested?

Reply:
Updated as part of below comment.